synapseclient/client.py

```diff
@@ -952,6 +961,15 @@ def store(self, obj, *, createOrUpdate=True, forceVersion=True, versionLabel=Non
             bundle = self._getEntityBundle(entity)
 
             if bundle:
+                if createOrUpdate:
+                    # update our properties from the existing bundle so that we have
```

Comment:
This would not work if we deleted some annotation entries since we would end up merging the ones we deleted back in.

Reply:
We shouldn't rely on the createOrUpdate flag being set to alter this behavior. I'm pretty sure most users don't even know that it exists.

Reply:
Changed it to only merge annotations when resolving a 409.